ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/population/internal/PopulationServiceImpl.java

```diff
@@ -232,22 +285,103 @@ public void resetPopulation(DataSet dataSet) {
 	 * 
 	 * @param group the group
 	 * @param groupDef the group entity definition
+	 * @param path A Child Context path
 	 */
-	private void addToPopulation(Group group, EntityDefinition groupDef) {
-		for (QName propertyName : group.getPropertyNames()) {
-			EntityDefinition propertyDef = AlignmentUtil.getChild(groupDef, propertyName);
+	private void addToPopulation(Group group, EntityDefinition groupDef, List<ChildContext> path) {
+		Iterable<? extends EntityDefinition> children = entityDefinitionService
+				.getChildren(groupDef);
+		if (children != null && children.iterator().hasNext()) {
+			for (EntityDefinition def : children) {
+				if (groupDef instanceof TypeEntityDefinition)
+					path = def.getPropertyPath();
+				evaluateContext(group, def, path);
+			}
+		}
+		else {
+			evaluateContext(group, groupDef, path);
+		}
+	}
+
+	private void evaluateContext(Group group, EntityDefinition groupDef, List<ChildContext> path) {
 
-			if (propertyDef != null) {
-				Object[] values = group.getProperty(propertyName);
+		if (path == null || path.isEmpty()) {
+			// group or instance at end of path
+			increase(groupDef, 1);
+		}
+		else {
+			ChildContext context = path.get(0);
+			List<ChildContext> subPath = null;
+			if (path.size() > 0) {
+				subPath = path.subList(1, path.size());
+			}
+			Object[] values = group.getProperty(context.getChild().getName());
+			if (values != null) {
+				// apply the possible source contexts
+				if (context.getIndex() != null) {
+					// select only the item at the index
+					int index = context.getIndex();
+					if (index < values.length) {
+						values = new Object[] { values[index] };
+					}
+					else {
+						values = new Object[] {};
+					}
+				}
+				if (context.getCondition() != null) {
+					// select only values that match the condition
+					List<Object> matchedValues = new ArrayList<Object>();
+					for (Object value : values) {
+						if (AlignmentUtil.matchCondition(context.getCondition(), value, group)) {
+							matchedValues.add(value);
+						}
+					}
+					values = matchedValues.toArray();
+				}
 
-				increase(propertyDef, values.length);
+				if (context.getChild().getName().equals(groupDef.getDefinition().getName())) {
+					increase(groupDef, values.length);
+				}
 
 				for (Object value : values) {
 					if (value instanceof Group) {
-						addToPopulation((Group) value, propertyDef);
+						addToPopulation((Group) value, groupDef, subPath);
```

Comment:
Here you pass all Group values on, but shouldn't that be done only for those that were counted for this level?
For instance: If it did not match the condition at this level, it should not be counted on the next.